### ISSUE TITLE
errors arised when build docker image

### ISSUE THREAD

**Issue by shanekong** [NVIDIA/TensorRT-LLM#82]
i followed the instruct to download the code:

# TensorRT-LLM uses git-lfs, which needs to be installed in advance.
apt-get update && apt-get -y install git git-lfs

git clone https://github.com/NVIDIA/TensorRT-LLM.git
cd TensorRT-LLM
git submodule update --init --recursive
git lfs install
git lfs pull

and then followed the instruct to build image:
docker build --pull  \
             --target devel \
             --file docker/Dockerfile.multi \
             --tag tensorrt_llm/devel:latest \
             .

then errors arised:
Sending build context to Docker daemon  549.5MB
Step 1/22 : ARG BASE_IMAGE=nvcr.io/nvidia/pytorch
Step 2/22 : ARG BASE_TAG=23.08-py3
Step 3/22 : FROM ${BASE_IMAGE}:${BASE_TAG} as base
23.08-py3: Pulling from nvidia/pytorch
Digest: sha256:12a39f22d6e3a3cfcb285a238b6219475181672ff41a557a75bdeeef6d630740
Status: Image is up to date for nvcr.io/nvidia/pytorch:23.08-py3
 ---> 9ff159797e3d
Step 4/22 : ENV BASH_ENV="/tmp/bash_env"
 ---> Running in 1f52361987e2
Removing intermediate container 1f52361987e2
 ---> 5c0d4a6994ca
Step 5/22 : SHELL ["/bin/bash", "-c"]
 ---> Running in f5b00f975d6e
Removing intermediate container f5b00f975d6e
 ---> 22b842bd9252
Step 6/22 : FROM base as devel
 ---> 22b842bd9252
Step 7/22 : COPY docker/common/install_base.sh install_base.sh
 ---> 0760de5497c8
Step 8/22 : RUN bash ./install_base.sh && rm install_base.sh
 ---> Running in 2d9df1b8701b
++ grep -oP '(?<=^ID=).+' /etc/os-release
++ tr -d '"'
+ ID=ubuntu
+ case "$ID" in
+ init_ubuntu
+ apt-get update
Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Get:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]
Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Reading package lists...
W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://security.ubuntu.com/ubuntu jammy-security InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://security.ubuntu.com/ubuntu jammy-security InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-updates InRelease' is not signed.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-backports InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-backports InRelease' is not signed.
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code

**Comment by sleepwalker2017** [comment]
I also meet this issue. It's weird!

**Comment by sleepwalker2017** [comment]
```shell
DOCKER_BUILDKIT=1 sudo docker build -t triton_trt_llm -f dockerfile/Dockerfile.trt_llm_backend .
Sending build context to Docker daemon  550.1MB
Step 1/34 : ARG BASE_IMAGE=nvcr.io/nvidia/tritonserver
Step 2/34 : ARG BASE_TAG=23.08-py3
Step 3/34 : FROM ${BASE_IMAGE}:${BASE_TAG} as base
 ---> 3e96065a3dcc
Step 4/34 : RUN apt-get update
 ---> Running in 5341b72fc2cb
Get:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1581 B]
Err:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease
  At least one invalid signature was encountered.
Get:2 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Err:3 http://security.ubuntu.com/ubuntu jammy-security InRelease
  At least one invalid signature was encountered.
Err:2 http://archive.ubuntu.com/ubuntu jammy InRelease
  At least one invalid signature was encountered.
Get:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Err:4 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  At least one invalid signature was encountered.
Get:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]
Err:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  At least one invalid signature was encountered.
Reading package lists...
W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease: At least one invalid signature was encountered.
E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease' is not signed.
W: GPG error: http://security.ubuntu.com/ubuntu jammy-security InRelease: At least one invalid signature was encountered.
E: The repository 'http://security.ubuntu.com/ubuntu jammy-security InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: At least one invalid signature was encountered.
E: The repository 'http://archive.ubuntu.com/ubuntu jammy InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-updates InRelease: At least one invalid signature was encountered.
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-updates InRelease' is not signed.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy-backports InRelease: At least one invalid signature was encountered.
E: The repository 'http://archive.ubuntu.com/ubuntu jammy-backports InRelease' is not signed.
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
```

**Comment by sleepwalker2017** [comment]
I cleaned some unused docker images and run it again, this error occurs:
```
DOCKER_BUILDKIT=1 sudo docker build -t triton_trt_llm -f dockerfile/Dockerfile.trt_llm_backend .
Sending build context to Docker daemon  550.1MB
Step 1/34 : ARG BASE_IMAGE=nvcr.io/nvidia/tritonserver
Step 2/34 : ARG BASE_TAG=23.08-py3
Step 3/34 : FROM ${BASE_IMAGE}:${BASE_TAG} as base
 ---> 3e96065a3dcc
Step 4/34 : RUN apt-get update
 ---> Running in b1245a9ffeed
Get:1 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1581 B]
Get:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [555 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Get:6 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]
Get:7 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [44.0 kB]
Get:8 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
Get:9 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [1131 kB]
Get:10 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [1305 kB]
Get:11 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1009 kB]
Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
Get:13 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
Get:14 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [1330 kB]
Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [49.8 kB]
Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1398 kB]
Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1274 kB]
Get:19 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [78.3 kB]
Get:20 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [32.6 kB]
Fetched 28.5 MB in 4s (6810 kB/s)
Reading package lists...
E: Problem executing scripts APT::Update::Post-Invoke-Success '/usr/bin/test -e /usr/share/dbus-1/system-services/org.freedesktop.PackageKit.service && /usr/bin/test -S /var/run/dbus/system_bus_socket && /usr/bin/gdbus call --system --dest org.freedesktop.PackageKit --object-path /org/freedesktop/PackageKit --timeout 4 --method org.freedesktop.PackageKit.StateHasChanged cache-update > /dev/null; /bin/echo > /dev/null'
E: Sub-process returned an error code
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
```

**Comment by Kelang-Tian** [comment]
Same error. You can modify it like this to solve the problem:
in docker/Dockerfile.multi
`ARG BASE_TAG=23.08-py3` -> `ARG BASE_TAG=22.12-py3`

**Comment by shanekong** [comment]
> Same error. You can modify it like this to solve the problem: in docker/Dockerfile.multi `ARG BASE_TAG=23.08-py3` -> `ARG BASE_TAG=22.12-py3`

I have tried to change the base ARG BASE_TAG=23.08-py3 -> ARG BASE_TAG=22.12-py3；
but the following step arise some other errors.
# To build the TensorRT-LLM code.
python3 ./scripts/build_wheel.py --trt_root /usr/local/tensorrt

can you get through the following steps sucessfully?

**Comment by shanekong** [comment]
when change the base ARG BASE_TAG=23.08-py3 -> ARG BASE_TAG=22.12-py3, i can build the image sucessfully. 

but error airsed when build the tensorTR-LLM code
python3 ./scripts/build_wheel.py --trt_root /usr/local/tensorrt

...
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 620; error   : Vector qualifier is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 620; error   : Operation .max requires .u32 or .s32 or .u64 or .s64 type for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 620; error   : Vector operand is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 620; error   : Vector operand is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 1909; warning : Vector Type not specified properly
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 1909; error   : Vector qualifier is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 1909; error   : Operation .max requires .u32 or .s32 or .u64 or .s64 type for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 1909; error   : Arguments mismatch for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 1909; error   : Arguments mismatch for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 5395; error   : Vector qualifier is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 5395; error   : Operation .max requires .u32 or .s32 or .u64 or .s64 type for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 5395; error   : Vector operand is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 5395; error   : Vector operand is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 8963; warning : Vector Type not specified properly
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 8963; error   : Vector qualifier is not allowed for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 8963; error   : Operation .max requires .u32 or .s32 or .u64 or .s64 type for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 8963; error   : Arguments mismatch for instruction 'atom'
ptxas /tmp/tmpxft_00000488_00000000-6_cudaFp8Utils.compute_90.ptx, line 8963; error   : Arguments mismatch for instruction 'atom'
ptxas fatal   : Ptx assembly aborted due to errors
make[3]: *** [tensorrt_llm/common/CMakeFiles/common_src.dir/build.make:188: tensorrt_llm/common/CMakeFiles/common_src.dir/cudaFp8Utils.cu.o] Error 255
...
[ 97%] Building CUDA object tensorrt_llm/kernels/CMakeFiles/kernels_src.dir/weightOnlyBatchedGemv/weightOnlyBatchedGemvBs4Int4b.cu.o
[ 98%] Building CUDA object tensorrt_llm/kernels/CMakeFiles/kernels_src.dir/weightOnlyBatchedGemv/weightOnlyBatchedGemvBs4Int8b.cu.o
[ 98%] Built target kernels_src
make[1]: *** [CMakeFiles/Makefile2:684: tensorrt_llm/CMakeFiles/tensorrt_llm.dir/rule] Error 2
make: *** [Makefile:179: tensorrt_llm] Error 2
Traceback (most recent call last):
  File "./scripts/build_wheel.py", line 248, in <module>
    main(**vars(args))
  File "./scripts/build_wheel.py", line 152, in main
    build_run(
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'cmake --build . --config Release --parallel 64 --target tensorrt_llm tensorrt_llm_static nvinfer_plugin_tensorrt_llm th_common ' returned non-zero exit status 2.
@Kelang-Tian, do you meet the same problem?

**Comment by Kelang-Tian** [comment]
I have built tensorrt-llm succeed with CUDA12.2 in H800.
Be sure to use the cuda12.2 version. Some cuda syntax will report errors on lower versions.

**Comment by Shixiaowei02** [comment]
It looks like this is a bug in a specific version of the docker program and described [here](https://github.com/moby/moby/pull/42681) instead of the ubuntu mirror. A complete solution to the problem might be to upgrade docker itself.

**Comment by Shixiaowei02** [comment]
> i followed the instruct to download the code:
> 
> # TensorRT-LLM uses git-lfs, which needs to be installed in advance.
> apt-get update && apt-get -y install git git-lfs
> 
> git clone https://github.com/NVIDIA/TensorRT-LLM.git cd TensorRT-LLM git submodule update --init --recursive git lfs install git lfs pull
> 
> and then followed the instruct to build image: docker build --pull --target devel --file docker/Dockerfile.multi --tag tensorrt_llm/devel:latest .
> 
> then errors arised: Sending build context to Docker daemon 549.5MB Step 1/22 : ARG BASE_IMAGE=nvcr.io/nvidia/pytorch Step 2/22 : ARG BASE_TAG=23.08-py3 Step 3/22 : FROM ${BASE_IMAGE}:${BASE_TAG} as base 23.08-py3: Pulling from nvidia/pytorch Digest: sha256:12a39f22d6e3a3cfcb285a238b6219475181672ff41a557a75bdeeef6d630740 Status: Image is up to date for nvcr.io/nvidia/pytorch:23.08-py3 ---> 9ff159797e3d Step 4/22 : ENV BASH_ENV="/tmp/bash_env" ---> Running in 1f52361987e2 Removing intermediate container 1f52361987e2 ---> 5c0d4a6994ca Step 5/22 : SHELL ["/bin/bash", "-c"] ---> Running in f5b00f975d6e Removing intermediate container f5b00f975d6e ---> 22b842bd9252 Step 6/22 : FROM base as devel ---> 22b842bd9252 Step 7/22 : COPY docker/common/install_base.sh install_base.sh ---> 0760de5497c8 Step 8/22 : RUN bash ./install_base.sh && rm install_base.sh ---> Running in 2d9df1b8701b ++ grep -oP '(?<=^ID=).+' /etc/os-release ++ tr -d '"'
> 
> * ID=ubuntu
> * case "$ID" in
> * init_ubuntu
> * apt-get update
>   Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
>   Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
>   Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
>   Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
>   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   Get:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]
>   Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
>   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
>   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
>   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   Reading package lists...
>   W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: GPG error: http://security.ubuntu.com/ubuntu jammy-security InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   E: The repository 'http://security.ubuntu.com/ubuntu jammy-security InRelease' is not signed.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   E: The repository 'http://archive.ubuntu.com/ubuntu jammy InRelease' is not signed.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: GPG error: http://archive.ubuntu.com/ubuntu jammy-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   E: The repository 'http://archive.ubuntu.com/ubuntu jammy-updates InRelease' is not signed.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
>   W: GPG error: http://archive.ubuntu.com/ubuntu jammy-backports InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
>   E: The repository 'http://archive.ubuntu.com/ubuntu jammy-backports InRelease' is not signed.
>   E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/_.deb /var/cache/apt/archives/partial/_.deb /var/cache/apt/*.bin || true'
>   E: Sub-process returned an error code

It looks like this is a bug in a specific version of the docker program and described [here](https://github.com/moby/moby/pull/42681) instead of the ubuntu mirror. A complete solution to the problem might be to upgrade docker itself. @shanekong
